vanilla-framework/vanilla-framework-react
# Major codebase cleanup

## Done

- Removed a ton of bloat that I think was introduced when this project was bootstrapped with `create-react-app` (at least I think it was). Since this is a component library, and not an app, a huge portion of that code is unnecessary, including:
  - Webpack configs. We only really need webpack in this project to run Storybook, which comes pre-baked with its own configs. The only thing I kept was the Friendly Errors Plugin.
  - Custom scripts in the scripts/ folder. `build` has been removed in favour of a `prepublish` script (running babel), `start` has been removed entirely since we use `start-storybook`, and `test` has been removed in favour of Jest CLI.
  - Added a `prepublish` script that simply uses Babel to transpile the code and put it in a dist folder. The files are no longer bundled, and they follow the same flow as the src folder (just without stories or test files)
  - Removed a bunch of packages that were no longer used with all the removed code.

## QA

- Pull code
- Check that `./run` still works 
- Open http://localhost:8102, make some changes to some src/lib/components files and see if hot module replacement still works
- Check that `./run test` still works
- Edit some test files to fail and see if the tests catch it
- Link the project to your global npm modules by running `npm link`
- Open a React project (you can use https://github.com/canonical-webteam/react-starter) and run `npm link vanilla-framework-react` (make sure you remove vanilla-framework-react from dependencies)
- See if the project still imports `vanilla-framework-react` properly

## TODO

- This is a first pass at packaging the project properly. In the future I think adding `rollup` as the bundler would be a good idea - general Internet consensus seems to be to use `rollup` for libs, `webpack` for apps

## Issues

Fixes #128
